programs/x509/cert_write.c

```diff
@@ -446,8 +446,8 @@ int load_pubkey_from_remote( const char * remote_info, mbedtls_pk_context * ctx
 {
     int key_idx = 0, offset = 0, ret = 0;
     const char * serial_port = NULL;
-    unsigned char func_buffer[10];
-    unsigned char pub_key_buf[100];
+    unsigned char func_buffer[2];   /* Op code: 1 + key Id: 1 */
+    unsigned char pub_key_buf[65];  /* ECDSA Pub key: 64 + ASN.1 overhead: 1 */
```

Comment:
65 is not right. ASN.1 is type-length-value so the overhead is always at least 2 bytes (type, length). For a 256-bit curve, the maximum size is: 2 * (32 + 1 + (1+1)) + (1+1) where 32 = each integer, 1 = leading 0 in the INTEGER encoding and 1+1 = type+length if length ≤ 127. That's a total of 72.

Reply:
This is strange. The ECDSA public key that goes in the certificate is 64+1 as I have seen so far.

Reply:
Please see here https://github.com/mazimkhan/mbedtls/blob/development/library/ecp.c#L480
This function only adds a tag ```0x04``` in front of the buffer before putting concatenated PK. It does not explain why it does so.

Reply:
Ah! That's because it isn't encoded in ASN.1. That's a different format which is defined in [SEC1](http://www.secg.org/sec1-v2.pdf) §2.3.3–2.3.4. Here the size is indeed 1 + 2 * B where B is the curve byte size, so 65 is correct.

Reply:
Updated comment. No change in code. CI already passed. Please consider merging.